libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorAdaptiveRecognizer.cs

```diff
@@ -0,0 +1,280 @@
+        /// </value>
+        [JsonProperty("modelPath")]
+        public StringExpression ModelPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the full path to the snapshot to use.
+        /// </summary>
+        /// <value>
+        /// Snapshot path.
+        /// </value>
+        [JsonProperty("snapshotPath")]
+        public StringExpression SnapshotPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the entity recognizers.
+        /// </summary>
+        /// <value>
+        /// The entity recognizers.
+        /// </value>
+        [JsonProperty("entityRecognizers")]
+#pragma warning disable CA2227 // Collection properties should be read only (keeping this consistent with RegexRecognizer)
+        public List<EntityRecognizer> EntityRecognizers { get; set; } = new List<EntityRecognizer>();
+#pragma warning restore CA2227 // Collection properties should be read only
+
+        /// <summary>
+        /// Gets or sets the disambiguation score threshold.
+        /// </summary>
+        /// <value>
+        /// Recognizer returns ChooseIntent (disambiguation) if other intents are classified within this score of the top scoring intent.
+        /// </value>
+        [JsonProperty("disambiguationScoreThreshold")]
+        public NumberExpression DisambiguationScoreThreshold { get; set; } = 0.05F;
+
+        /// <summary>
+        /// Gets or sets detect ambiguous intents.
+        /// </summary>
+        /// <value>
+        /// When true, recognizer will look for ambiguous intents (intents with close recognition scores from top scoring intent).
+        /// </value>
+        [JsonProperty("detectAmbiguousIntents")]
+        public BoolExpression DetectAmbiguousIntents { get; set; } = false;
+
+        /// <summary>
+        /// Return recognition results.
+        /// </summary>
+        /// <param name="dialogContext">Context object containing information for a single turn of conversation with a user.</param>
+        /// <param name="activity">The incoming activity received from the user. The Text property value is used as the query text for QnA Maker.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <param name="telemetryProperties">Additional properties to be logged to telemetry with the LuisResult event.</param>
+        /// <param name="telemetryMetrics">Additional metrics to be logged to telemetry with the LuisResult event.</param>
+        /// <returns>A <see cref="RecognizerResult"/> containing the QnA Maker result.</returns>
+        public override async Task<RecognizerResult> RecognizeAsync(DialogContext dialogContext, Schema.Activity activity, CancellationToken cancellationToken, Dictionary<string, string> telemetryProperties = null, Dictionary<string, double> telemetryMetrics = null)
+        {
+            var text = activity.Text ?? string.Empty;
```

Comment:
"text" is never used.

Reply:
Removed.

Reply:
I still see it.

---
In reply to: [464716181](https://github.com/microsoft/botbuilder-dotnet/pull/4375#discussion_r464716181) [](ancestors = 464716181)